contracts/modules/oracle/ReadonlyOracle.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "chainlink/interfaces/FeedRegistryInterface.sol";
+import {Denominations} from "chainlink/Denominations.sol";
+import "../../interfaces/IOracle.sol";
+
+/**
+ * @title   - Readonly version Chainlink Feed Registry Wrapper
+ * @notice  - simple contract that wraps Chainlink's Feed Registry to get asset prices for any tokens without needing to know the specific oracle address
+ *          - only makes request for USD prices and returns results in standard 8 decimals for Chainlink USD feeds
+ *
+ */
+contract ReadonlyOracle is IOracle {
+    /// @notice registry - Chainlink Feed Registry with aggregated prices across
+    FeedRegistryInterface public immutable registry;
+    /// @notice NULL_PRICE - null price when asset price feed is deemed invalid
+    int256 public constant NULL_PRICE = 0;
+    /// @notice PRICE_DECIMALS - the normalized amount of decimals for returned prices in USD
+    uint8 public constant PRICE_DECIMALS = 8;
+    /// @notice MAX_PRICE_LATENCY - amount of time between oracle responses until an asset is determined toxiz
+    /// Assumes Chainlink updates price minimum of once every 24hrs and 1 hour buffer for network issues
+    uint256 public constant MAX_PRICE_LATENCY = 25 hours;
+
+    constructor(address _registry) {
+        registry = FeedRegistryInterface(_registry);
+    }
+
+    /**
+     * @param token - ERC20 token to get USD price for
+     * @return price - the latest price in USD to 8 decimals
+     */
+    function getLatestAnswer(address token) external view returns (int256) {
```

Comment:
@kibagateaux we cannot call it `_getLAtestAnswer` as it differs from the `IOracle` function name. We can either omit inheriting from `IOracle` or leave it the same. Up to you.

Reply:
Ah i wanted the function in the same oracle contract. U can just copypasta this code over. Its just their for cheaper/easier reads onchain + offchain if someone doesnt care about all the additional security checks we do